packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -991,17 +991,19 @@ class FileImage extends ImageProvider<FileImage> {
   Future<ui.Codec> _loadAsync(FileImage key, DecoderBufferCallback? decode, DecoderCallback? decodeDeprecated) async {
     assert(key == this);
 
-    final Uint8List bytes = await file.readAsBytes();
-    if (bytes.lengthInBytes == 0) {
+    final int lengthInBytes = file.lengthSync();
+    if (lengthInBytes == 0) {
```

Comment:
Is there test coverage of the `== 0` case?

Reply:
It appears that one of the failures is from such a case:
```
03:52 +6199 ~13 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/painting/image_provider_test.dart: File image with empty file throws expected error and evicts from cache [E]                          
  Expected: <true>
    Actual: <false>
  
  package:test_api                                    expect
  package:flutter_test/src/widget_tester.dart 460:16  expect
  test/painting/image_provider_test.dart 77:5         main.<fn>
```

Reply:
Yeah, we have a test that covers this. Interestingly if fails if I use the sync length method: https://github.com/flutter/flutter/issues/113044